Update the harness documentation.

ElliotGregory opened the pull request:
Update the harness documentation to include AMQP and code for new Flood tests

ElliotGregory committed:
Update ReadMe.md

Commit:
Update floodPublisher to have a message limiter (-mr)

For the FloodPublier/Subscriber not to overflow the message queues the
publisher need to have a rate limiter which is just greater than the
subscriber can consume the messages.

Commit:
FloodPublisher can limit the rate message are generated

For the FloodPublisher not to overload the queues the rate that the
message are generated need to be slightly more than the consuming rate.

Commit:
Merge branch 'master' of https://github.com/ElliotGregory/perf-harness.git

Commit:
Update documentation to include AMQP Harness

MartinRoss-IBM commented on AMQPPerfHarness/src/com/ibm/uk/hursley/perfharness/amqp/FloodPublisher.properties:
```diff
@@ -7,10 +7,13 @@
 tpt.desc = Topic per thread
 tpt.type = java.lang.Boolean
+
+mr.dflt = 
```
There is already an -rt flag available for the WorkerThread that FloodPublisher implements which provides rate control. Does this already provide the functionality you are looking to provide here?

ElliotGregory replied:
As time has marched by could I suggest the pull proceed with the attributes as is. MQ performance team is wanting to have access to my changes. Possibly they could review/combine these attributes.

MartinRoss-IBM commented on AMQPPerfHarness/src/com/ibm/uk/hursley/perfharness/amqp/FloodPublisher.java:
```diff
@@ -98,10 +121,42 @@ public boolean oneIteration() throws Exception {
 		else {
 			client.sendMessage("flood", topic, message);
 		}
+		messageRateControl ();
```
Please see comment against the properties file relating to the existing -rt flag

MartinRoss-IBM commented:
@ElliotGregory: Not sure if you were notified automatically of my comment above regarding the existing -rt flag so just @ ing you.

ElliotGregory commented on AMQPPerfHarness/src/com/ibm/uk/hursley/perfharness/amqp/FloodPublisher.properties:
```diff
@@ -7,10 +7,13 @@
 tpt.desc = Topic per thread
 tpt.type = java.lang.Boolean
+
+mr.dflt = 
```
Regarding the  -mr attribute  and the existing -rt, you are right, I will look in to it.

MartinRoss-IBM commented on MQJavaPerfHarness/.classpath:
```diff
@@ -1,20 +1,8 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Could you remove this change from the pull request

MartinRoss-IBM replied:
Perhaps we should add these .classpath files to the .gitignore...

MartinRoss-IBM commented on JMSPerfHarness/.classpath:
```diff
@@ -1,13 +1,9 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Could you remove this change from the pull request?

MartinRoss-IBM replied:
Perhaps we should add these .classpath files to the .gitignore...

MartinRoss-IBM commented on PerfHarness/.classpath:
```diff
@@ -1,6 +1,6 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Could you remove this change from the pull request

MartinRoss-IBM replied:
Perhaps we should add these .classpath files to the .gitignore...

MartinRoss-IBM commented on JMSPerfHarness/.gitignore:
```diff
@@ -1,2 +1,4 @@
 perfharness.jar
 VERSION.txt
+.gitignore
+/bin/
```
This doesn't seem to have worked... there are class files included in the change request within the bin directories, this may be due to the leading '/'

ElliotGregory replied:
Changed

MartinRoss-IBM commented on ReadMe.md:
```diff
@@ -43,6 +43,18 @@ Building Perfharness in Eclipse requires some extra dependancies importing to th
+
+* The allclient.jar is required from the a MQ V9.0 installation.
+
+* Locate all the pre-req jars in located in /lib of the AMQPPerfharness.
```
I don't think that we are able to include these JAR files here... instead we typically need to point to where to acquire them such that a user can accept any associated license

ElliotGregory replied:
I think there a confusion here so I have rewords this section. I was instructing the end user to find the allclients.jar from a MQ V9 installation and copy into /lib.

Commit:
# Remove the XMl header from the .classpath
#
# Committer: ElliotGregory <elliotgregory@dhcp-9-20-235-158.hursley.ibm.com>
#
# On branch master
# Your branch is up-to-date with 'origin/master'.
#
# Changes to be committed:
#	modified:   .classpath
#

Commit:
# Remove XML header from .classpath file and reword AMQP sevtion in ReadMe.md.
#
# Committer: ElliotGregory <egregory@uk.ibm.com>
#
# On branch master
# Your branch is up-to-date with 'origin/master'.
#
# Changes to be committed:
#	modified:   JMSPerfHarness/.classpath
#	modified:   JMSPerfHarness/.gitignore
#	modified:   MQJavaPerfHarness/.classpath
#	modified:   PerfHarness/.classpath
#	modified:   ReadMe.md
#
# Changes not staged for commit:
#	modified:   ReadMe.md
#
# Untracked files:
#	.DS_Store
#	AMQPPerfHarness/.DS_Store
#	AMQPPerfHarness/src/.DS_Store
#	AMQPPerfHarness/src/com/.DS_Store
#	AMQPPerfHarness/src/com/ibm/.DS_Store
#	AMQPPerfHarness/src/com/ibm/uk/.DS_Store
#	AMQPPerfHarness/src/com/ibm/uk/hursley/.DS_Store
#

Commit:
# Remove XML header from .classpath file and reword AMQP sevtion in ReadMe.md.
#
# Committer: ElliotGregory <egregory@uk.ibm.com>
#
# On branch master
# Your branch is up-to-date with 'origin/master'.
#
# Changes to be committed:
#	modified:   JMSPerfHarness/.classpath
#	modified:   JMSPerfHarness/.gitignore
#	modified:   MQJavaPerfHarness/.classpath
#	modified:   PerfHarness/.classpath
#	modified:   ReadMe.md
#
# Changes not staged for commit:
#	modified:   ReadMe.md
#
# Untracked files:
#	.DS_Store
#	AMQPPerfHarness/.DS_Store
#	AMQPPerfHarness/src/.DS_Store
#	AMQPPerfHarness/src/com/.DS_Store
#	AMQPPerfHarness/src/com/ibm/.DS_Store
#	AMQPPerfHarness/src/com/ibm/uk/.DS_Store
#	AMQPPerfHarness/src/com/ibm/uk/hursley/.DS_Store
#

Commit:
#  Backout recent changes
#
# Committer: ElliotGregory <egregory@uk.ibm.com>
#
# On branch master
# Your branch is up-to-date with 'origin/master'.
#
# Changes to be committed:
#	modified:   ../HTTPPerfHarness/.classpath
#	modified:   ../JMSPerfHarness/.classpath
#	modified:   ../MQJavaPerfHarness/.classpath
#	modified:   ../PerfHarness/.classpath
#
# Changes not staged for commit:
#	modified:   ../ReadMe.md
#
# Untracked files:
#	../.DS_Store
#	.DS_Store
#	src/.DS_Store
#	src/com/.DS_Store
#	src/com/ibm/.DS_Store
#	src/com/ibm/uk/.DS_Store
#	src/com/ibm/uk/hursley/.DS_Store
#

Commit:
#  Backout recent changes
#
# Committer: ElliotGregory <egregory@uk.ibm.com>
#
# On branch master
# Your branch is up-to-date with 'origin/master'.
#
# Changes to be committed:
#	modified:   ../HTTPPerfHarness/.classpath
#	modified:   ../JMSPerfHarness/.classpath
#	modified:   ../MQJavaPerfHarness/.classpath
#	modified:   ../PerfHarness/.classpath
#
# Changes not staged for commit:
#	modified:   ../ReadMe.md
#
# Untracked files:
#	../.DS_Store
#	.DS_Store
#	src/.DS_Store
#	src/com/.DS_Store
#	src/com/ibm/.DS_Store
#	src/com/ibm/uk/.DS_Store
#	src/com/ibm/uk/hursley/.DS_Store
#

MartinRoss-IBM reviewed (approved):
Approving changes and will raise an issue to to utilise the -rt flag
